contrib/pkg/awsprivatelink/endpointvpc/add.go

```diff
@@ -0,0 +1,469 @@
+				default:
+					log.WithError(err).Fatal("Failed to authorize traffic from the associated VPC's worker SG to the endpoint VPC's default SG")
+				}
+			}
+
+			log.Info("Authorizing traffic from the endpoint VPC's default SG to the associated VPC's worker SG")
+			if _, err = awsutils.AuthorizeAllIngressFromSG(
+				associatedVpcClients,
+				aws.String(associatedVpcWorkerSG),
+				aws.String(endpointVPCDefaultSG),
+				aws.String(fmt.Sprintf("Access from default SG of endpoint VPC %s", o.endpointVpcId)),
+			); err != nil {
+				// Proceed if ingress already authorized, fail otherwise
+				switch aerr, ok := err.(awserr.Error); {
+				case ok && aerr.Code() == "InvalidPermission.Duplicate":
+					log.Warnf("Traffic from the endpoint VPC's default SG to the associated VPC's worker SG is already authorized")
+				default:
+					log.WithError(err).Fatal("Failed to authorize traffic from the endpoint VPC's default SG to the associated VPC's worker SG")
+				}
+			}
+
+		// Associated VPC & endpoint VPC in different regions => allow ingress from CIDR of the peer
+		default:
+			log.Info("Authorizing traffic from the associated VPC's CIDR block to the endpoint VPC's default SG")
+			if _, err = awsutils.AuthorizeAllIngressFromCIDR(
+				o.endpointVpcClients,
+				aws.String(endpointVPCDefaultSG),
+				associatedVpcCIDR,
+				aws.String(fmt.Sprintf("Access from CIDR block of associated VPC %s", associatedVpcId)),
+			); err != nil {
+				// Proceed if ingress already authorized, fail otherwise
+				switch aerr, ok := err.(awserr.Error); {
+				case ok && aerr.Code() == "InvalidPermission.Duplicate":
+					log.Warnf("Traffic from the associated VPC's CIDR block to the endpoint VPC's default SG is already authorized")
+				default:
+					log.WithError(err).Fatal("Failed to authorize traffic from the associated VPC's CIDR block to the endpoint VPC's default SG")
+				}
+			}
+
+			log.Info("Authorizing traffic from the endpoint VPC's CIDR block to the associated VPC's worker SG")
+			if _, err = awsutils.AuthorizeAllIngressFromCIDR(
+				associatedVpcClients,
+				aws.String(associatedVpcWorkerSG),
+				endpointVpcCIDR,
+				aws.String(fmt.Sprintf("Access from CIDR block of endpoint VPC %s", o.endpointVpcId)),
+			); err != nil {
+				// Proceed if ingress already authorized, fail otherwise
+				switch aerr, ok := err.(awserr.Error); {
+				case ok && aerr.Code() == "InvalidPermission.Duplicate":
+					log.Warnf("Traffic from the endpoint VPC's CIDR block to the associated VPC's worker SG is already authorized")
+				default:
+					log.WithError(err).Fatal("Failed to authorize traffic from the endpoint VPC's CIDR block to the associated VPC's worker SG")
+				}
+			}
+		}
+	}
+
+	// Update HiveConfig
+	o.addEndpointVpcToHiveConfig()
+
+	return nil
+}
+
+func (o *endpointVPCAddOptions) addEndpointVpcToHiveConfig() {
+	log.Infof("Adding endpoint VPC %v to HiveConfig", o.endpointVpcId)
+
+	// Get HiveConfig
+	hiveConfig := &hivev1.HiveConfig{}
```

Comment:
We're getting the HiveConfig in `Complete()` -- can we just add it to `o` there and reuse it here?

Reply:
Yep. Fixed. 

I also made some changes to remove.go to reuse HiveConfig as well as endpointVpcIdx (index of the endpoint VPC to remove in the list of endpoint VPC inventory).